From 755ddb9745a5abf6e191654e8f95fb4aa87fea09 Mon Sep 17 00:00:00 2001
From: Chuanxiao Dong <chuanxiao.dong@intel.com>
Date: Tue, 29 Jul 2014 14:48:44 +0800
Subject: [PATCH 07/15] mmc: fix eMMC High speed disable failure issue

This bug was introduced in the v3.4 kernel mmc driver.
It forces set eMMC High speed support even the Host
controller caps has been set to not support High speed.
This patch add check of the host controller caps before
setting the High speed support.

Change-Id: Ie90a7dfc16b58bf47aeca14c8ef66dd6ab8cb969
Signed-off-by: Chuanxiao Dong <chuanxiao.dong@intel.com>
---
 drivers/mmc/core/mmc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/core/mmc.c b/drivers/mmc/core/mmc.c
index 22080fe0dcfb..eae167eb8fcc 100644
--- a/drivers/mmc/core/mmc.c
+++ b/drivers/mmc/core/mmc.c
@@ -1096,7 +1096,7 @@ static int mmc_init_card(struct mmc_host *host, u32 ocr,
 				mmc_card_set_hs200(card);
 				mmc_set_timing(card->host,
 					       MMC_TIMING_MMC_HS200);
-			} else {
+			} else if (host->caps & MMC_CAP_MMC_HIGHSPEED) {
 				mmc_card_set_highspeed(card);
 				mmc_set_timing(card->host, MMC_TIMING_MMC_HS);
 			}
-- 
1.9.1

